scripts/Jenkinsfiles/quality

```diff
@@ -0,0 +1,219 @@
+                }
+            }
+        }
+    }
+    post {
+        always {
+            unstash 'quality-1-reports'
+            unstash 'quality-2-reports'
+            unstash 'quality-3-reports'
+            unstash 'quality-4-reports'
+            // Check for warnings
+            warnings canComputeNew: false, canResolveRelativePaths: false, canRunOnFailed: true, categoriesPattern: '',
+                defaultEncoding: '', excludePattern: '', healthy: '', includePattern: '', messagesPattern: '',
+                parserConfigurations: [[parserName: 'Pep8', pattern: 'reports/pep8/pep8.report'], [parserName: 'Pep8', pattern: 'reports/**/pylint.report']],
+                unHealthy: ''
+            // Publish Quality report
+            publishHTML([allowMissing: true, alwaysLinkToLastBuild: false, keepAll: true,
+                reportDir: 'reports/metrics/',
+                reportFiles: 'pylint/*view*/,pep8/*view*/,python_complexity/*view*/,xsscommitlint/*view*/,xsslint/*view*/,eslint/*view*/',
+                reportName: 'Quality Report', reportTitles: ''])
+
+            script{
+                if (env.ghprbPullId != null) {
+                    // For PR jobs, run the edx-platform-test-notifier for PR reporting
+                    build job: 'edx-platform-test-notifier', parameters: [string(name: 'PR_NUMBER', value: "${ghprbPullId}")], wait: false
+                } else {
+                    // For master jobs run github-build-status and report to slack when necessary
+                    if (currentBuild.currentResult == "SUCCESS") {
+                        create_deployment = "true"
+                        build_status = "success"
+                        build_description = "Build Passed"
+                    }
+                    else {
+                        create_deployment = "false"
+                        build_status = "failure"
+                        build_description = "Build Failed"
+                    }
+
+                    commit_sha = sh(returnStdout: true, script: 'git rev-parse HEAD').trim()
+                    build job: 'github-build-status', parameters: [string(name: 'GIT_SHA', value: commit_sha),
+                        string(name: 'GITHUB_ORG', value: 'edx'), string(name: 'GITHUB_REPO', value: 'edx-platform'),
+                        string(name: 'TARGET_URL', value: "https://build.testeng.edx.org/job/${JOB_NAME}/${BUILD_NUMBER}/"),
+                        string(name: 'DESCRIPTION', value: build_description), string(name: 'CONTEXT', value: 'jenkins/quality-pipeline-master'),
+                        string(name: 'CREATE_DEPLOYMENT', value: create_deployment), string(name: 'BUILD_STATUS', value: build_status)],
+                        propagate: false, wait: false
+                    if (currentBuild.currentResult != "SUCCESS"){
+                        slackSend "${currentBuild.currentResult}: `${JOB_NAME}` #${BUILD_NUMBER}:\n${BUILD_URL}"
+                    }
```

Comment:
This block here ^ used to trigger the github-build-status job will be common to all of the 'master' jobs. Can we pull it out into a separate file so that they can all use it or does that go against the Jenkinsfile pattern?

Reply:
Sorting out common code with these seems to be a bit of a pain. @jmbowman and I talked about it a little bit here: https://github.com/edx/edx-platform/pull/18038#discussion_r185305815 .

Official docs here: https://jenkins.io/doc/book/pipeline/shared-libraries/

Not saying we can't do this. But it's not super trivial.

Reply:
That sounds like a good idea. I definitely don't want to derail us getting off of build-flow by worrying about DRY-ness, but we should follow up on that in the future for sure, because I can see drift being an issue.